Source/Core/Core/NetPlayServer.cpp

```diff
@@ -981,16 +976,18 @@ void NetPlayServer::unmapPortThread()
 // discovers the IGD
 bool NetPlayServer::initUPnP()
 {
+  static bool s_inited = false, s_error = false;
+
   std::vector<UPNPDev*> igds;
```

Comment:
You may want to move these variables after the init and error checking, otherwise the vector gets constructed and then immediately discarded if any of the checks are true.